static/js/chart/draw.ts

```diff
@@ -427,10 +434,207 @@ function drawLineChart(
   }
 }
 
+
+/**
+ * Return all styles information for given dataGroupsDict.
+ * {
+ *     colors: [],
+ *     dashes: [],
+ *     statVars: [],
+ *     geoIds: [],
+ *     maxV: ,
+ * }
+ *
+ * @param dataGroupsDict
+ */
+function generateStyles(
+    dataGroupsDict: {[geoId: string]: DataGroup[] },
+) {
+  let styles = {};
+  styles["colors"] = [];
+  styles["dashes"] = [];
+  styles["statVars"] = [];
+  styles["geoIds"] = [];
+
+  let dataGroups : DataGroup[];
+  dataGroups = Object.values(dataGroupsDict)[0];
+  let legendText = dataGroups.map((dataGroup) => dataGroup.label ? dataGroup.label: '');
+  let colorFn = getColorFn(legendText);
+
+  for (let i = 0; i < dataGroups.length; i++) {
+    styles["colors"].push(colorFn(dataGroups[i].label));
+    styles["statVars"].push(dataGroups[i].label);
+  }
+
+  let dashIndex = 0;
+  let maxV = 0;
+  for (let geoId in dataGroupsDict) {
+    dataGroups = dataGroupsDict[geoId];
+    maxV = Math.max(maxV, Math.max(...dataGroupsDict[geoId].map((dataGroup) => dataGroup.max())));
+    styles["geoIds"].push(geoId);
+    styles["dashes"].push(DASHES[dashIndex]);
+    dashIndex++;
+  }
+  styles["maxV"] = maxV;
+
+  return styles;
+}
+
+
+/**
+ * Draw a group of lines chart with in-chart legend given a dataGroupsDict with different geoIds.
+ *
+ * @param id: DOM id.
+ * @param width: width for the chart.
+ * @param height: height for the chart.
+ * @param dataGroupsDict: {[geoId: string]: DataGroup[]}.
+ * @param unit
+ */
+function drawGroupLineChart(
+    id: string,
+    width: number,
+    height: number,
+    dataGroupsDict: {[geoId: string]: DataGroup[] },
+    unit?: string
+) {
+  // Get all styles.
+  let styles = generateStyles(dataGroupsDict);
+
+  let dataGroups: DataGroup[];
+  dataGroups = Object.values(dataGroupsDict)[0];
+
+  let minV = 0;
+  let maxV = styles["maxV"];
```

Comment:
these two can be generated from a function like: computeValueRange()

Reply:
Done